src/System.Management.Automation/help/HelpFileHelpProvider.cs

```diff
@@ -368,7 +368,7 @@ internal Collection<string> GetExtendedSearchPaths()
                     {
                         // Get all the directories under the module path
                         // * and SearchOption.AllDirectories gets all the version directories.
-                        string[] directories = Directory.GetDirectories(psModulePath, "*", SearchOption.AllDirectories);
+                        IEnumerable<string> directories = Directory.EnumerateDirectories(psModulePath, "*", SearchOption.AllDirectories);
```

Comment:
```suggestion
                        Ivar possibleModuleDirectories = Directory.EnumerateDirectories(psModulePath, "*", SearchOption.AllDirectories).Where(static directory => !ModuleUtils.IsPossibleResourceDirectory(directory));
```

Would this be better to take advantage of using items in the enumerable before the whole search is complete.

Reply:
Signed off either way.

Reply:
> Would this be better to take advantage of using items in the enumerable before the whole search is complete.

The collection is not iterated until the `foreach` as Linq queries are evaluated using deferred execution. So the suggested change has no effect